dartagnan/src/main/java/com/dat3m/dartagnan/encoding/WmmEncoder.java

```diff
@@ -151,26 +136,20 @@ public BooleanFormula encodeRelations(SolverContext ctx) {
     }
 
     // Encodes all axioms. This should be called after <encodeRelations>
-    public BooleanFormula encodeConsistency(SolverContext ctx) {
+    public BooleanFormula encodeConsistency() {
         checkInitialized();
         logger.info("Encoding consistency");
-        final BooleanFormulaManager bmgr = ctx.getFormulaManager().getBooleanFormulaManager();
+        final BooleanFormulaManager bmgr = context.getFormulaManager().getBooleanFormulaManager();
         return memoryModel.getAxioms().stream()
                 .filter(ax -> !ax.isFlagged())
-                .map(ax -> ax.consistent(ax.getRelation().getEncodeTupleSet(), analysisContext, ctx))
+                .map(ax -> ax.consistent(ax.getRelation().getEncodeTupleSet(), context))
                 .reduce(bmgr.makeTrue(), bmgr::and);
     }
 
     private final class RelationEncoder implements Relation.Visitor<BooleanFormula> {
-        final ExecutionAnalysis exec = analysisContext.get(ExecutionAnalysis.class);
-        final AliasAnalysis alias = analysisContext.get(AliasAnalysis.class);
-        final SolverContext ctx;
         final BooleanFormulaManager bmgr;
-        final IntegerFormulaManager imgr;
-        public RelationEncoder(SolverContext c) {
-            ctx = c;
-            bmgr = c.getFormulaManager().getBooleanFormulaManager();
-            imgr = c.getFormulaManager().getIntegerFormulaManager();
+        public RelationEncoder() {
+            bmgr = context.getFormulaManager().getBooleanFormulaManager();
```

Comment:
I have seen this access quite a few times now.
I think a direct getter `context.getBmgr()` would make it a little easier :).

Reply:
No problem, but I am opposed to abbreviations in public methods.

Reply:
Sure, you can give it the longer name.